### ISSUE TITLE
Parallel manager seems to not work with em_http and typhoeus adapters

### ISSUE THREAD

**Issue by bogdanRada** [lostisland/faraday#560]
I am not sure where to ask this, but i can't seem to use faraday with em_http adapter. i keep getting errors like **NoMethodError: undefined method `add' for true:TrueClass**  in gems/faraday-0.9.2/lib/faraday/adapter/em_http.rb:97:in  method **perform_request**  and another error  **NoMethodError: undefined method `finish' for nil:NilClass** in file   gems/faraday-0.9.2/lib/faraday/adapter/em_http.rb:118:in `block in perform_request' because it seems option **parallel_manager** is either sometimes NIL or TRUE , instead of being a instance of Faraday::Adapter::EMHttp::Manager. 

I am currently using this:

```ruby
client = Asana::Client.new do |c|
        c.authentication :access_token, ENV['ASANA_API_KEY']
        c.debug_mode
        c.faraday_adapter :em_http
        c.configure_faraday do |conn|
          conn.use Faraday::Adapter::EMHttp
          conn.request  :url_encoded         
          conn.response :logger
          conn.default_parallel_manager = Faraday::Adapter::EMHttp.setup_parallel_manager
        end
      end
client.tasks.create(workspace: ENV['ASANA_WORKSPACE_ID'], name: "Hello world!", assignee: "me")
```

Maybe i am doing something wrong? Even if i remove the last line from the **configure_faraday** i get same errors. Could you help me please? 

I even tried this:

```ruby
conn.parallel_manager = Faraday::Adapter::EMHttp::Manager.new
```
but that seems to disable entirely the parallel manager 

Could you tell me what i am doing wrong?

**Comment by bogdanRada** [comment]
any update on this? Is this in future plans to be fixed?

**Comment by bogdanRada** [comment]
is this repository still maintained?

**Comment by 987poiuytrewq** [comment]
I think the configuration block should look like below. I think you're trying to use an adapter as a middleware which isn't quite correct. 

The `c.faraday_adapter` seems to get passed down to faraday as `builder.adapter` (https://github.com/Asana/ruby-asana/blob/master/lib/asana/http_client.rb#L137)

You don't need to specify a parallel manager unless you want to use a custom one. I personally use the typhoeus adapter with no issues.

```
client = Asana::Client.new do |c|
        c.authentication :access_token, ENV['ASANA_API_KEY']
        c.debug_mode
        c.faraday_adapter :em_http
        c.configure_faraday do |conn|
          conn.request  :url_encoded         
          conn.response :logger
        end
      end
```

**Comment by bogdanRada** [comment]
I have tried to use typhoeus . but unfortunately when doing parallel requests i get error " Too many requests" although there are never more than 4. 
I tried using em-synchrony but that does not work either as it suffers from same errors as for em-http-request.  


After all the testing i eventually gave up and now i am using em-http-request directly which works ok but i would really want to use the Asana gem instead of doing work arounds . 

I really need this to get fixed. I could help also. I just need some guidance in the right direction. 
 Any help would be much appreciated
